### ISSUE TITLE
Timeout while deploying

### ISSUE THREAD

**Issue by ngmiduc** [google-github-actions/deploy-cloud-functions#333]
### TL;DR

Hi, 

I am not sure what I am doing wrong, 
but I am getting a timeout while deploying.

It seems to run but it takes a very long time until it timeouts.

You can find the infos below.

### Expected behavior

Should complete deployment

### Observed behavior

Returned error

### Action YAML

```yaml
name: ...

on:
  push:
    paths: ...
    branches: ...
    
jobs:
  jobid:
    name: Deployment
    runs-on: ubuntu-latest
    steps:
      - name: Checkout Repo
        uses: actions/checkout@v3

      - id: auth
        uses: google-github-actions/auth@v0
        with:
          credentials_json: ${{ secrets.GCP_KEY }}

      - id: deploy
        uses: google-github-actions/deploy-cloud-functions@v0
        with:
          name: <name of existing cloud function in GCP>
          runtime: python38
          entry_point: main (function is called like that in pythonfile.py>
          memory_mb: 256MB
          region: ...
          env_vars: ENV1=1,ENV2=2
          source_dir: subfolder/subsubfolder
          project_id: ...
          description: ....
          build_environment_variables: GOOGLE_FUNCTION_SOURCE=pythonfile.py
          secret_environment_variables: SECRET=projects/.../secret:latest
          min_instances: 0
          max_instances: 1
          event_trigger_type: google.pubsub.topic.publish
          event_trigger_resource: projects/.../topics/topicId
          event_trigger_service: pubsub.googleapis.com
          deploy_timeout: 500

```
```


### Log output

```text
Creating new Cloud Function revision
Deploying Cloud Function
Still deploying Cloud Function (1/n)
Still deploying Cloud Function (2/n)
Still deploying Cloud Function (3/n)
Still deploying Cloud Function (4/n)
Still deploying Cloud Function (5/n)
Still deploying Cloud Function (6/n)
Still deploying Cloud Function (7/n)
Still deploying Cloud Function (8/n)
Still deploying Cloud Function (9/n)
Still deploying Cloud Function (10/n)
Still deploying Cloud Function (11/n)
Still deploying Cloud Function (12/n)
...
Still deploying Cloud Function (59/n)
Error: google-github-actions/deploy-cloud-functions failed with: operation timed out
```
```


### Additional information

_No response_

**Comment by sethvargo** [comment]
Hi @ngmiduc 

This usually indicates a problem during your deployment, which is only visible in the deployment or runtime logs for the Cloud Function deployment. The most common cause is because the runtime service account doesn't have permissions to access the secrets. Unfortunately we can't access or display those logs in our output.

**Comment by ngmiduc** [comment]
Hi,
thanks for your reply. 

Okay after I increased the timeout it got this:

`Error: google-github-actions/deploy-cloud-functions failed with: operation failed: Function deployment failed due to a health check failure. This usually indicates that your code was built successfully but failed during a test execution. Examine the logs to determine the cause. Try deploying again in a few minutes if it appears to be transient. This deployment uses Secrets. Ensure that the runtime service account 'pinkdolphin-prod-4541@appspot.gserviceaccount.com' has the permission 'roles/secretmanager.secretAccessor' on the project or secrets.`

**Comment by ngmiduc** [comment]
Looks like maybe a problem with the secrets

**Comment by ngmiduc** [comment]
Confirmed.
I used a secret with the suffix `:latest` but I think I need to remove that. Otherwise it will deploy `:latest:latest`

**Comment by sethvargo** [comment]
Hi @ngmiduc 

The input for secrets specifies a secret _version_ (see https://github.com/google-github-actions/deploy-cloud-functions#inputs). So the value should be:

```text
SECRET=projects/.../secrets/my-secret/versions/123 # or latest
```

**Comment by ngmiduc** [comment]
Ah okay!
Yeah I see the syntax now.
Thanks
